### PR TITLE
[Clang-Tidy] Tighten clang-tidy warnings into errors for bugprone.

### PR DESCRIPTION
**Pull request by Skylion007**
## Motivation and Context

* There were a few annoying RAII warnings and bugprone-branch clones that were being emitted as warnings instead of errors. It seems like the RAII warnings are false positives though since Caffe2 has similar nolints. False positive occurs when a module is constructed with nodefs, which usually is an error, but in these few cases is not. When it is not an error, nolint comments should be explicit.
* Now the main warning left in the code base is implicit narrowing conversions.

## How Has This Been Tested
* Locally and on CircleCI

## Types of changes

- [X] Docs change / refactoring / dependency upgrade
-
## Checklist

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have completed my CLA (see **CONTRIBUTING**)
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/esp/bindings/AttributesManagersBindings.cpp [hunk 1]
```diff
@@ -274,12 +274,14 @@ void initAttributesManagersBindings(py::module& m) {
 
   // ==== Light Layout Attributes Template manager ====
   declareBaseAttributesManager<LightLayoutAttributes>(m, "BaseLightLayout");
+  // NOLINTNEXTLINE(bugprone-unused-raii)
```
Comment (Skylion007) [inline src/esp/bindings/AttributesManagersBindings.cpp:277]: Would like some confirmations from the reviewers that these are false positives.

Comment (erikwijmans) [reply src/esp/bindings/AttributesManagersBindings.cpp:277]: Yeah, those are all false positives.  We should probably turn this off globally for src/esp/bindings.  The most common use of RAII is locking mutexs and there likely won't ever be any mutexs in bindings.

Comment (Skylion007) [reply src/esp/bindings/AttributesManagersBindings.cpp:277]: Unfortunately, disabling a single check from clang-tidy in a subfolder is rather difficult unless we upgrade to clang-tidy 11 which supports inheriting from the parent config. As such, I'll just leave them as warnings.

**Review by jturner65 — APPROVED** [review]
LGTM

**Pull request merged**